### ISSUE TITLE
Option to disable font ligature

### ISSUE THREAD

**Issue by sonulohani** [cpeditor/cpeditor#433]
Font ligature makes code more readable however that's based on preferences. Some people like it and some people [hate ](http://www.chicagowebexperts.com/blog/im-not-a-fan-of-programming-font-ligatures) it. However in cpeditor we force user to use font ligature which is not good. We are refraining developer to use their favorite fonts with their preferences. 

To solve this problem, I would like to suggest a feature requests where user will get an option to enable/disable font ligature same like how we have in visual studio code.

**Comment by sonulohani** [comment]
Let me know if you agree with the suggestions. I could work on this feature.

**Comment by coder3101** [comment]
@all-contributors please add @sonulohani for ideas

**Comment by coder3101** [comment]
> Let me know if you agree with the suggestions. I could work on this feature.

Yes, that will be a nice addition. How do you plan to achieve it?

**Comment by sonulohani** [comment]
> 
> 
> > Let me know if you agree with the suggestions. I could work on this feature.
> 
> Yes, that will be a nice addition. How do you plan to achieve it?

> I was looking at [QGlyphRun](https://doc.qt.io/qt-5/qglyphrun.html) class to achieve this.

**Comment by sonulohani** [comment]
Hi @coder3101 

Seems like QGlyphRun will not work in this case. I checked kde editor Kate/Kdevelop and saw that even they dont have an option to disable font ligatures. I tried using stylesheet and didn't work as well. We might need to put this in backlog for now.

**Comment by neko-para** [comment]
@sonulohani It seems that the problem is the QCodeEditor cannot provide an option for this. So you can add an issue in cpeditor/QCodeEditor. After that issue closed, we can restart thinking about this issue, and I believe it's simple to add an option for that.

**Comment by ouuan** [comment]
QCodeEditor is maintained by us, and I think it's easier to track issues if all issues are in the same repo.

**Comment by ouuan** [comment]
I believe "awating on dependency" is for the dependencies not maintained by us, so that we can wait for it and it will be solved by others. We can't wait for a "dependency" which is not maintained by others.

**Comment by coder3101** [comment]
"awaiting on editor" is a new label applies to issues that can be fixed if changes are required in QCodeEditor.

You can easily filter on this label to see all issues/enhancement for Editor.

**Comment by ouuan** [comment]
"help wanted" is not "we don't know how to implement this, can anyone help", it's the issues which are harder than the good first issues, but not too hard for new contributors, and not assigned to any maintainer.
